Functions/Quantification/xASL_quant_External.m

```diff
@@ -0,0 +1,706 @@
+
+				% We go through the LabDurs, PLDs, and TEs and keep grouping similar LD+PLD with increasing TE into groups
+				LabDursGrouped = [];
+				PLDsGrouped = [];
+				nTEGrouped = [];
+				iTE = 1;
+
+				while iTE<=length(TEs)
+					% We define a block
+					iTEstart = iTE;
+					iTEend   = iTE;
+
+					% We enlarge the block if PLD and LD are similar, but TE increasing
+					while (iTEend+1 <= length(TEs)) &&... % There are more values
+							(LabDurs(iTEend+1)==LabDurs(iTEstart)) && (PLDs(iTEend+1)==PLDs(iTEstart)) &&... % The further PLDs and LD are matching the block
+							(TEs(iTEend+1)>TEs(iTEend)) % And TE increases
+						% If all fulfilled, then we increase the block
+						iTEend = iTEend+1;
+					end
+					% We add the block to the grouped vector
+					LabDursGrouped = [LabDursGrouped, LabDurs(iTEstart)];
+					PLDsGrouped = [PLDsGrouped, PLDs(iTEstart)];
+					nTEGrouped = [nTEGrouped, iTEend-iTEstart+1];
+					iTE = iTEend+1;
+				end
+
+				LabDurs = LabDursGrouped;
+				PLDs = PLDsGrouped;
+				nTE = nTEGrouped;
+			case 'vaby'
+				TEs = round(jsonPWI4D.Q.EchoTime'/1000,3);
+				nTE = length(TEs);
+			case 'basil'
+				% Normal multi-timepoint without multi-TE
+				TEs = [];
+				nTE = [];
+		end
+
+		switch (lower(strQuantificationType))
+			case 'fabber'
+				% Printing the values in the FSL option file (PLD=ti, LD=tau)
+				% If we have for a give PLD more TEs, then we print once the PLD, once nTE for each collection of multi-TE volumes
+				for iPLD = 1:length(PLDs)
+					fprintf(FIDoptionFile, '--ti%d=%.2f\n', iPLD, PLDs(iPLD) + LabDurs(iPLD));
+				end
+				for iNTE = 1:length(nTE)
+					fprintf(FIDoptionFile, '--nte%d=%d\n', iNTE, nTE(iNTE)); % --nte1=8 --nte2=8 --nte3=8 (if nTE=8)
+				end
+
+				if length(nTE) == 1 && nTE == 1
+					% For a single-TE, we have to repeat it for each volume
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(1));
+					end
+				else
+					% For multi-TE, we print all of them
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(iTE));
+					end
+				end
+
+				% Future extension - specify the repetitions explicitly
+				%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+				%fprintf(FIDoptionFile, '--repeats=1\n');
+			case 'vaby'
+				% Printing the values in the commandline (PLD=plds, LD=taus)
+				for iPLD = 1:length(PLDs)
+					if iPLD == 1
+						ExternalOptions = [ExternalOptions ' --plds=' sprintf('%.2f', PLDs(iPLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', PLDs(iPLD))];
+					end
+				end
+
+				for iTE = 1:length(TEs) %So for each volume, we print a TE value
+					if iTE == 1
+						ExternalOptions = [ExternalOptions ' --tes=' sprintf('%.3f', TEs(iTE))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.3f', TEs(iTE))];
+					end
+				end
+			case 'basil'
```

Comment:
They are extremely secretive about the settings and I can't configure FABBER to run with full BASIL functionality. So I mainly use FABBER for DEBBIE because it has a completely different module and uses very different config parameters. So lets keep it as it is.